crates/bin/src/entry.rs

```diff
@@ -117,6 +120,22 @@ pub fn install_crates(
         client,
         gh_api_client,
         jobserver_client,
+        crates_io_rate_limit: {
```

Comment:
Happy for that to happen in another PR but I think API wise this is something that should be essentially hardcoded in binstalk, not something that a consumer can reduce or bypass, as it's a hard policy from crates.io

Reply:
I've replaced this field with a new type `CratesIoRateLimit` that only exposes `<CratesIoRateLimit as Default>::default` as public and async fn `CratesIoRateLimit::tick` as `pub(crate)`.

I thought about making all fields of `opts::Options` private and instead exposing a function `opts::Options::new`, but that also looks ugly to me: with that many arguments/parameters and some of them even have the same type (e.g. `bool`, `PathBuf`), it's hard to get the order of params right.

So instead I use a newtype for this.